pkgs/gnolang/precompile.go

```diff
@@ -105,29 +105,29 @@ func PrecompileAndCheckMempkg(mempkg *std.MemPackage) error {
 	return nil
 }
 
-func Precompile(source string, tags string, filename string) (string, error) {
+func Precompile(source string, tags string, filename string) (*ast.File, string, error) {
```

Comment:
What do you think about defining a return struct for this method, like `PrecompileResult`? It makes it cleaner to modify the return values on this method, so it doesn't return more than 2 values like now:

`func Precompile(...) (PrecompileResult, error)`

Reply:
Resolving this convo as you've rewritten the implementation